### PR TITLE
etlegacy: fix and refactor

### PR DESCRIPTION
**Pull request by drupol**
Hello,

I'm unable to run `etlegacy` on my machine, I get the following error log:

```
❯ etl.x86_64
tty]*** buffer overflow detected ***: terminated
*** buffer overflow detected ***: terminated
DOUBLE SIGNAL FAULT: Received signal 6, exiting...
*** buffer overflow detected ***: terminated
DOUBLE SIGNAL FAULT: Received signal 6, exiting...
*** buffer overflow detected ***: terminated
DOUBLE SIGNAL FAULT: Received signal 6, exiting...
*** buffer overflow detected ***: terminated
DOUBLE SIGNAL FAULT: Received signal 6, exiting...
...
DOUBLE SIGNAL FAULT: Received signal 6, exiting...
*** buffer overflow detected ***: terminated
DOUBLE SIGNAL FAULT: Received signal 6, exiting...
*** buffer overflow detected ***: terminated
fish: Job 1, 'etl.x86_64' terminated by signal SIGSEGV (Address boundary error)
~ ✘
```

This PR:

- [x] Added the `etl` and `etlded` binaries
- [x] Disable `fortify` to have fully working binaries and fix the aforementioned issue
- [x] Remove unused `runCommand` argument
- [x] Add myself as maintainer
- [x] Reduce amount of local vars
- [x] Refactor `fetchAsset` function
- [x] Build server and client
- [x] Update `cmakeFlags` and change default install directories

Ping @ashleyghooper 

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by bjornfor** [comment]
> etlegacy: remove mainProgram attribute 

It's not removing it though, just refactoring.

**Review by ashleyghooper — CHANGES_REQUESTED** [review]
Hi @drupol 
thanks for your suggested changes. Many of them look okay, and adding the dedicated server is a good move, but I'd like to discuss a few of them. See comments in the review.

Also, I'm keen to hear your thoughts on how we support for other platforms than Linux `x86_64`. Especially Linux `i686` and Darwin.

File: pkgs/games/etlegacy/default.nix [hunk 1]
```diff
@@ -1,138 +1,134 @@
...
+    makeBinaryWrapper
+  ];
 
-    preBuild = ''
-      # Required for build time to not be in 1980
-      export SOURCE_DATE_EPOCH=$(date +%s)
-      # This indicates the build was by a CI pipeline and prevents the resource
-      # files from being flagged as 'dirty' due to potentially being custom built.
-      export CI="true"
-    '';
```
[the remark this answers is not in the document]
Comment (drupol) [reply pkgs/games/etlegacy/default.nix:102]: Fair enough, going to revert my change.

File: pkgs/games/etlegacy/default.nix [hunk 2]
```diff
@@ -1,138 +1,134 @@
...
+
+  postInstall = ''
+    ln -s ${pak0} $out/lib/etlegacy/etmain/pak0.pk3
+    ln -s ${pak1} $out/lib/etlegacy/etmain/pak1.pk3
+    ln -s ${pak2} $out/lib/etlegacy/etmain/pak2.pk3
+
+    makeWrapper $out/lib/etlegacy/etl.* $out/bin/etl
+    makeWrapper $out/lib/etlegacy/etlded.* $out/bin/etlded
+
+    rm -rf $out/share/applications/com.etlegacy.ETLegacy.x86_64.desktop
```
[the remark this answers is not in the document]
Comment (drupol) [reply pkgs/games/etlegacy/default.nix:105]: You're right, I've skipped the `x86_64` in the filename, going to make an adjustment.

File: pkgs/games/etlegacy/default.nix [hunk 3]
```diff
@@ -1,138 +1,134 @@
...
-  fakeGit = writeScriptBin "git" ''
-    #! ${stdenv.shell} -e
-    if [ "$1" = "describe" ]; then
-      echo "${version}"
-    fi
-  '';
-  mainProgram =
-    if stdenv.hostPlatform.system == "i686-linux"
-    then "etl.i386"
-    else "etl.x86_64";
```
[the remark this answers is not in the document]
Comment (drupol) [reply pkgs/games/etlegacy/default.nix:66]: Fair enough, going to leave the `.<arch>` in the next iteration and use their desktop file.

File: pkgs/games/etlegacy/default.nix [hunk 4]
```diff
@@ -1,138 +1,134 @@
...
-    {
-      asset = "pak2.pk3";
-      sha256 = "a48ab749a1a12ab4d9137286b1f23d642c29da59845b2bafc8f64e052cf06f3e";
-    };
-  fakeGit = writeScriptBin "git" ''
-    #! ${stdenv.shell} -e
-    if [ "$1" = "describe" ]; then
-      echo "${version}"
-    fi
-  '';
```
[the remark this answers is not in the document]
Comment (drupol) [reply pkgs/games/etlegacy/default.nix:62]: Going to restore this as well.

**Comment by drupol** [comment]
Thanks for the detailed review, I'm going to start working on them in one hour.

**Comment by drupol** [comment]
I think I've addressed everything, could you please check again?

File: pkgs/games/etlegacy/default.nix (hunk 1 above)
[the remark this answers is not in the document]
Comment (ashleyghooper) [reply pkgs/games/etlegacy/default.nix:102]: Thanks for that. Previously I think there was a suggestion we could talk to upstream about getting an upstream change to make this unnecessary. I guess it's there to make it clear to people who compile themselves and might have modified things that they're running a version that could potentially be unstable when playing online.

**Review by ashleyghooper — APPROVED** [review]

**Comment by drupol** [comment]
Youhou ! thanks !

**Pull request merged**